File: tests/test-drift/src/main/java/pize/tests/drift/Drift.java
package pize.tests.drift;

import pize.Pize;
import pize.app.AppAdapter;
import pize.graphics.camera.CenteredOrthographicCamera;
import pize.graphics.gl.Gl;
import pize.graphics.texture.Texture;
import pize.graphics.util.batch.TextureBatch;
import pize.io.glfw.Key;

public class Drift extends AppAdapter{

    public static void main(String[] args){
        Pize.create("Drift", 2000, 1000);
        Pize.run(new Drift());
    }


    private final Car car;
    private final TextureBatch batch;
    private final CenteredOrthographicCamera camera;
    private final Texture roadTexture;

    public Drift(){
        car = new Car();
        batch = new TextureBatch();
        camera = new CenteredOrthographicCamera();
        roadTexture = new Texture("road.png");
    }

    public void update(){
        if(Key.ESCAPE.isDown())
            Pize.exit();

        car.update();

        camera.position.set(car.getPosition());//.rotDeg(-car.getRotation()).add(0, 130);
        //camera.setRotation(-car.getRotation());
        camera.setScale(0.4F);
        camera.update();
    }

    public void render(){
        Gl.clearColorBuffer();
        Gl.clearColor(0.2, 0.2, 0.3, 1);
        batch.begin(camera);

        batch.draw(roadTexture, -200, -2000, 400, 4000);
        car.render(batch);

        batch.end();
    }

    public void resize(int width, int height){
        camera.resize(width, height);
    }

    public void dispose(){
        car.dispose();
        batch.dispose();
        roadTexture.dispose();
    }

}

File: tests/test-drift/src/main/kotlin/pize/tests/drift/Drift.kt
package pize.tests.drift

import pize.Pize.create
import pize.Pize.exit
import pize.Pize.run
import pize.app.AppAdapter
import pize.graphics.camera.CenteredOrthographicCamera
import pize.graphics.gl.Gl.clearColor
import pize.graphics.gl.Gl.clearColorBuffer
import pize.graphics.texture.Texture
import pize.graphics.util.batch.TextureBatch
import pize.io.glfw.Key

class Drift : AppAdapter() {
    private val car: Car
    private val batch: TextureBatch
    private val camera: CenteredOrthographicCamera
    private val roadTexture: Texture

    init {
        car = Car()
        batch = TextureBatch()
        camera = CenteredOrthographicCamera()
        roadTexture = Texture("road.png")
    }

    override fun update() {
        if (Key.ESCAPE.isDown) exit()
        car.update()
        camera.position.set(car.position) //.rotDeg(-car.getRotation()).add(0, 130);
        //camera.setRotation(-car.getRotation());
        camera.scale = 0.4f
        camera.update()
    }

    override fun render() {
        clearColorBuffer()
        clearColor(0.2, 0.2, 0.3, 1.0)
        batch.begin(camera)
        batch.draw(roadTexture, -200f, -2000f, 400f, 4000f)
        car.render(batch)
        batch.end()
    }

    override fun resize(width: Int, height: Int) {
        camera.resize(width, height)
    }

    override fun dispose() {
        car.dispose()
        batch.dispose()
        roadTexture.dispose()
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            create("Drift", 2000, 1000)
            run(Drift())
        }
    }
}
